### ISSUE TITLE
Async code in `ProducesTickets` implementation

### ISSUE THREAD

**Issue by tvsfx** [rustls/rustls#1808]
**Checklist**
* [x] I've searched the issue tracker for similar requests

**Is your feature request related to a problem? Please describe.**
I am writing a custom implementation of the [`ProducesTickets`](https://github.com/rustls/rustls/blob/a8d3857106dd5933d52c8b49252fdee874fa3e5f/rustls/src/server/server_conn.rs#L74) trait. As part of encryption and decryption, I occasionally need to make some asynchronous database calls (when I need to rotate my STEK), which is currently not possible because the `encrypt`/`decrypt` methods are synchronous. 

**Describe the solution you'd like**
I was wondering if there are any plans to support this use case, or whether there's an existing solution for this that I'm missing.

**Describe alternatives you've considered**
As a workaround, I am thinking of spinning up a separate task to monitor the keys and rotate when necessary (to avoid calling `spawn_blocking`). I am not sure whether this is the best way to go about this issue.

**Comment by cpu** [comment]
> I was wondering if there are any plans to support this use case, or whether there's an existing solution for this that I'm missing.

I think https://github.com/rustls/rustls/issues/850 is the most relevant issue for capturing plans in this area. There's also the start of some work in https://github.com/rustls/rustls/pull/1648, but only the client-side and so nothing in relation to `ProducesTickets` yet.

**Comment by djc** [comment]
Yeah, I think a potential workaround is to have some kind of channel to a background task.
